pandas/tseries/index.py

```diff
@@ -1507,6 +1508,16 @@ def date(self):
         """
         return self._maybe_mask_results(_algos.arrmap_object(self.asobject.values, lambda x: x.date()))
 
+    @property
+    def weekday_name(self):
+        """
+        Returns numpy array of weekday name string. The name of day in a week (ex: Friday)
+
+        .. versionadded:: 0.18.0
+        """
+        weekdays = [calendar.day_name[i].lower() for i in range(7)]
+        return self._maybe_mask_results(_algos.arrmap_object(self.asobject.values, lambda x: weekdays[x.weekday()]))
+
```

Comment:
You are going to want to do something like this as what you are doing is going to be much much slower (only reason is this is done for ``.date`` is that its creating python objects so no way around it).

```
In [25]: index = pd.date_range('20130101 09:00:00',periods=20)

In [26]: weekdays = [calendar.day_name[i].lower() for i in range(7)]

In [27]: pandas.core.common.take_nd(np.array(weekdays),index.weekday)
Out[27]: 
array(['tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday',
       'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday',
       'sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday',
       'saturday', 'sunday'], 
      dtype='|S9')
```

then return this an an object array
you *may* need to do ``com._ensure_int64`` on the ``i.weekday`` (I don't remember)